File: Programmers/Lv.2/연속 부분 수열 합의 개수/131701.java
import java.util.*;

class Solution {
    public int solution(int[] elements) {
        int answer = 0;
        
        int[] arr = new int[elements.length * 2];
        for (int i = 0; i < elements.length; i++) {
            arr[i] = elements[i];
            arr[elements.length + i] = elements[i];
        }
        
        Set<Integer> set = new HashSet<>();
        for (int j = 0; j < elements.length; j++) {
            int sum = 0;
            for (int cnt = 0; cnt < elements.length; cnt++) {
                sum += arr[j + cnt];
                set.add(sum);
            }
        }
        
        answer = set.size();
        
        return answer;
    }
}

File: Programmers/Lv.2/연속 부분 수열 합의 개수/131701.js
function solution(elements) {
    let answer = 0;
    
    const set = new Set();
    const arr = [...elements, ...elements];
    
    for (let i = 0; i < elements.length; i++) {
        let sum = 0;
        for (let j = 0; j < elements.length; j++) {
            sum += arr[i + j];
            set.add(sum);
        }
    }
    
    answer = set.size;
    
    return answer;
}
